packages/playwright-core/src/client/network.ts

```diff
@@ -694,7 +711,7 @@ export class RouteHandler {
     if (this.noWaitOnUnrouteOrClose || noWait)
       handlerActivations.forEach(h => h.ignoreException = true);
     else
-      await Promise.all(handlerActivations.map(h => h.complete));
+      await Promise.all(handlerActivations.filter(h => !h.route._didTryToHandle).map(h => h.complete));
```

Comment:
- Should we set `ignoreException` flag for these? I'd prefer all invocations to be in one of the two possible states.
- What if my route does something after the `fulfill()`? I think we should only mark those that did try to handle and failed, not all of them.

Reply:
> * Should we set `ignoreException` flag for these? I'd prefer all invocations to be in one of the two possible states.

Route doesn't know anything about `handlerInvocation` structure. We can pass it an exception handler that will flip the flag and it's a property of the route rather than a particular call of a handler, so I don't quite see how we could reuse ignoreException flag for it.